frappe/public/js/frappe/views/treeview.js

```diff
@@ -94,6 +94,11 @@ frappe.views.TreeView = class TreeView {
 		this.page.main.addClass("frappe-card");
 
 		if (this.opts.show_expand_all) {
+
+			this.page.add_inner_button(__("Collapse All"), function () {
+				me.tree.collapse_all();
```

Comment:
```suggestion
				me.tree.load_children(me.tree.root_node, true);

```

All the code related to collapsing feels unnecessary. Why not just do this?

Reply:
Generic functionality > specific code. 

Especially if specific code is manipulating DOM 😄

Reply:
Oh right! I didn't realise that calling `load_children` again would toggle the root node. However, if the tree is not expanded - initial state - then the "Collapse" button would actually expand the tree right?

Reply:
It won't do anything if it's already collapsed AFAIK. You can try it out. 


Ideally if possible we shouldn't show these buttons where they don't make sense. 

Like showing "collapse all" when it's all collapsed already 👀

Reply:
> Like showing "collapse all" when it's all collapsed already 👀

Yeah that's what I wanted to do initially - but I wasn't able to 😅. I'll try that again.

I'll test it with the `load_children` function call if it's already collapsed - or better yet - if it is collapsed - I'll just not do anything (I was thinking I could add an alert but that would be too much I guess)?

Reply:
So `load_children` was loading the nodes if the root node was collapsed. 

I have changed the logic of the `collapse_all` function - it loops over all the nodes that have been loaded (so it does not fetch new nodes from the server) and then toggles only the ones that are expanded. Does this work?

Reply:
I think just using load_children with false is fine. A fully collapsed root is useless. 


The point of collapsing is to have at least 1st level open so the user can start from root nodes again.

Like why would I want to see this? This will require one more click to get root nodes. 

<img width="808" alt="Screenshot 2022-08-31 at 11 47 09 AM" src="https://user-images.githubusercontent.com/9079960/187607407-f98719c1-0ce8-4834-82d1-fbfc6b90d42c.png">

Reply:
Good point. I'll check if load_children collapses the nodes after level 2 or not. If it does, I'll use that, else I'll adjust the current logic to only collapse the level 2 and below nodes.

Reply:
I does, I checked and pushed :p